plugins/upload-assets-to-s3/src/tasks/upload-assets-to-s3.ts

```diff
@@ -53,6 +53,7 @@ export default class UploadAssetsToS3 extends Task<typeof UploadAssetsToS3Schema
         const data = await s3.upload(params).promise()
         this.logger.info(`Uploaded ${styles.filepath(filepath)} to ${styles.URL(data.Location)}`)
       } else {
+        const uploadPromises: Array<Promise<any>> = []
```

Comment:
**suggestion**: is it possible to type the `any` here given you are pushing a specific promise to the array on line 69?

Reply:
Yes! I have a question about this as it's my first time writing TS (exciting!). I think it would be  `Array<Promise<aws.S3.ManagedUpload.SendData>>>` but I wasn't sure if it should be more generic than that?

Reply:
I'm pretty sure you can steal the return type off of `s3.upload(params).promise` somehow. I hadn't thought about this before though, how typing the failure as well as the success works

Reply:
ahhh this is vaguely coming back to me: https://stackoverflow.com/questions/50071115/typescript-promise-rejection-type

I think you check the type of reject before you act upon it but you can't type it